soc/riscv/riscv-privilege/hi3861/Kconfig.defconfig.series

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2023 Chen Xingyu <hi@xingrz.me>
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_SERIES_HI3861
+
+config SOC_SERIES
+	default "hi3861"
+
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default 24000000
+
+config RISCV_SOC_INTERRUPT_INIT
+	default y
+
+config RISCV_HAS_CPU_IDLE
+	default y
+
+config RISCV_GP
+	default y
+
+config NUM_IRQS
+	int
+	default 62
+
+config FLASH_LOAD_OFFSET
+	default 0xd3c0
```

Comment:
Why is there a flash load offset here, does that mean if I have a bootloader at one address and an application at e.g. 0x20000, then instead of being placed at 0x20000 the application should be placed at 0x2d3c0?

Reply:
The boot process of this SoC is somewhat complex. Let me explain:

1. The **romboot**, which placed at a readonly part of the chip, verifies and copies the **flashboot** to `RAM_BASE + 160KB`, then boots it up.
2. The **flashboot**, located at `FLASH_BASE + 0x00`, is built with a [few lines of open-sourced code](https://github.com/xingrz/hal_hisilicon/blob/ce6606e1f2ad5a2fbb186f8d52984d34244ced54/hi3861/boot/flashboot/startup/main.c) and some close-sourced libs. It verifies and boots the kernel at `FLASH_BASE + 0xd3c0`.

This implies some limitations:

1. Some undocumented initializations are done in the flashboot.
2. The image loaded by the romboot is not XIP-compatible, and is limited in length to `RAM_SIZE - 160KB`, which is 120KB.
3. The offset `0xd3c0` acually includes `64 bytes` of the header of the flashboot image, `52 KB` reserved for the flashboot, and `960 bytes` of metadata verified by the flashboot.

Fortunately, since the flashboot is built from source in the `hal_hisilicon` module, we can reduce the offset, which is default to `0xd3c0` in the manufacturer-provided SDK, to at least 64 KB.

Reply:
> The flashboot, located at FLASH_BASE + 0x00, is built with a [few lines of open-sourced code](https://github.com/xingrz/hal_hisilicon/blob/ce6606e1f2ad5a2fbb186f8d52984d34244ced54/hi3861/boot/flashboot/startup/main.c) and some close-sourced libs. It verifies and boots the kernel at FLASH_BASE + 0xd3c0.

This presents a problem, I don't think we can accept this for an entire soc in the tree since without this whole soc is useless, the current use of "blobs" (which requires user interaction) is to add "binary blobs" but no current in-tree board needs them to be able to boot. @carlescufi seems like a TSC discussion would be needed?

Reply:
Indeed from https://docs.zephyrproject.org/latest/contribute/bin_blobs.html
> To mitigate this risk, the scope of upstream library blobs is limited. The project maintainers define an open source test suite that an upstream target must be able to pass using only open source software included in the mainline distribution and its modules. The open source test suite currently consists of:
>
>    samples/philosophers
>
>    tests/kernel

So if this board cannot boot and run both of the above without blobs then this soc cannot be included in zephyr.